snow/engine/snowman/transitive.go

```diff
@@ -1176,6 +1158,13 @@ func (t *Transitive) getProcessingAncestor(ctx context.Context, initialVote ids.
 	}
 }
 
+// canDependOn reports true if it is guaranteed for the provided block ID to
+// eventually either be fulfilled or abandoned.
+func (t *Transitive) canDependOn(blkID ids.ID) bool {
```

Comment:
This is probably a very dumb question but can you expand on why we shouldn't return that we can depend on blkIDs that are decided?

Reply:
If there is an invariant that it must be checked prior to calling this function that the block should not be decided, it may be worth adding a comment.

Reply:
> // canDependOn reports true if it is guaranteed for the provided block ID to
// eventually either be fulfilled or abandoned.

Decided blocks are not guaranteed to be either fulfilled or abandoned. So this function returns `false`. It is definitely possible that decided blockIDs are passed into this function.